c7n/policy.py

```diff
@@ -728,12 +715,56 @@ def is_lambda(self):
 
     def validate(self):
         m = self.get_execution_mode()
+        if m is None:
+            raise PolicyValidationError(
+                "Invalid Execution mode in policy %s" % (self.data,))
         m.validate()
         for f in self.resource_manager.filters:
             f.validate()
         for a in self.resource_manager.actions:
             a.validate()
 
+    def get_variables(self):
+        # Global policy variable expansion, we have to carry forward on
+        # various filter/action local vocabularies. Where possible defer
+        # by using a format string.
+        #
+        # See https://github.com/capitalone/cloud-custodian/issues/2330
+        return {
+            # standard runtime variables for interpolation
+            'account': '{account}',
+            'account_id': self.options.account_id,
+            'region': self.options.region,
+            # non-standard runtime variables from local filter/action vocabularies
+            #
+            # notify action
+            'policy': self.data,
+            'event': '{event}',
```

Comment:
this unfortunately may have some compatibility considerations, accessing keys of dictionaries in the format string won't work across this deferral mechanism where we pass in a simple string.

Reply:
the use cases for this seem pretty minimal, given its for notify and in notify you can use a full template engine with access to the same variables.